submissions/VBystrov/friends-app/js/index.js

```diff
@@ -0,0 +1,326 @@
+const endpoint = 'https://randomuser.me/api/';
+const includeFields = `inc=${[
+  'gender',
+  'name',
+  'location',
+  'picture',
+  'dob',
+  'email',
+].join(',')}`;
+const amountFriends = `results=${50}`;
+const options = `?${[includeFields, amountFriends].join('&')}`;
+const requestFriendsUrl = `${endpoint}${options}`;
+const initialSearchParameters = {
+  name: null,
+  age: null,
+  gender: 'all',
+  sorting: 'nameAZ',
+};
+
+class FriendsApp {
+  constructor() {
+    this.init(requestFriendsUrl);
+  }
+
+  init(url) {
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((friendsData) => {
+        const friendlist = new Friendlist(friendsData.results);
+        const search = new Search(friendlist);
+      })
+      .catch((error) => {
+        console.error(
+          'There has been a problem with your fetch operation:',
+          error
+        );
+      });
+  }
+}
+
+class Friendlist {
+  constructor(friendsData) {
+    this.container = document.getElementById('friendlist');
+    this.createCards(friendsData);
+  }
+
+  createCards(friendlistData) {
+    this.cards = [];
+    friendlistData.forEach((cardData) => {
+      this.cards.push(new Card(cardData));
+    });
+  }
+
+  applySearch(search) {
+    this.applyFilters(search);
+    this.applySorting(search.sorting);
+  }
+
+  applyFilters(search) {
```

Comment:
`search` quite opaque naming, TBH. Looks more like a state of some controllers, etc.